ads/google/a4a/utils.js

```diff
@@ -139,16 +142,27 @@ function buildAdUrl(
       {name: 'amp_v', value: '$internalRuntimeVersion$'},
       {name: 'd_imp', value: '1'},
       {name: 'dt', value: startTime},
+      {name: 'ifi', value: slotNumber},
       {name: 'adf', value: domFingerprint(adElement)},
-      {name: 'c', value: makeCorrelator(clientId, documentInfo.pageViewId)},
+      {name: 'c', value: getCorrelator(global, clientId)},
       {name: 'output', value: 'html'},
       {name: 'nhd', value: iframeDepth},
-      {name: 'eid', value: adElement.getAttribute('data-experiment-id')},
+      {name: 'iu', value: a4a.element.getAttribute('data-ad-slot')},
```

Comment:
I'll add a TODO. Yes, validateData is the right thing to use. So it ought to be moved somewhere to be used by both 3p and A4A. And someone needs to figure out which are required. (I used to know....)